File: MapGUI3/src/Model/expressions/ArithmeticExpression.java
package Model.expressions;

import Exceptions.InvalidOperationException;
import Model.Heap;
import Model.SymbolTable;
import Exceptions.CustomException;
import Exceptions.DivisionByZeroException;

public class ArithmeticExpression extends Expression {

    private Expression expr1;
    private Expression expr2;
    private char operation;

    public ArithmeticExpression(Expression expr1, char operation, Expression expr2) {
        this.expr1 = expr1;
        this.expr2 = expr2;
        this.operation = operation;
    }

    public int evaluate(SymbolTable<String, Integer> symTable, Heap ph) {
        if (operation == '+')
            return expr1.evaluate(symTable,ph) + expr2.evaluate(symTable,ph);
        else if (operation == '-')
            return expr1.evaluate(symTable,ph) - expr2.evaluate(symTable,ph);
        else if (operation == '*')
            return expr1.evaluate(symTable,ph) * expr2.evaluate(symTable,ph);
        else if (operation == '/') {
            int e1 = expr1.evaluate(symTable,ph);
            int e2 = expr2.evaluate(symTable,ph);
            if (e2 == 0) throw new DivisionByZeroException();
            return e1 / e2;
        }
        else
            throw new InvalidOperationException();
    }

    public String toString() {
        return "(" + expr1.toString() + operation + expr2.toString() + ")";
    }

}

File: MapA7/MapA7/Model/Expressions/ArithmeticExpression.cs
using MapA7.Model.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model.Expressions
{
    public class ArithmeticExpression:IExpression
    {
        private IExpression expression1;
        private IExpression expression2;
        private String op;

        public ArithmeticExpression(IExpression expression1, String op, IExpression expression2)
        {
            this.expression1 = expression1;
            this.expression2 = expression2;
            this.op = op;
        }

        public int Evaluate(SymbolTable symbolTable)
        {
            switch (op)
            {
                case "+":
                    return expression1.Evaluate(symbolTable) + expression2.Evaluate(symbolTable);
                case "-":
                    return expression1.Evaluate(symbolTable) - expression2.Evaluate(symbolTable);
                case "*":
                    return expression1.Evaluate(symbolTable) * expression2.Evaluate(symbolTable);
                case "/":
                    int e1 = expression1.Evaluate(symbolTable);
                    int e2 = expression2.Evaluate(symbolTable);
                    if (e2 != 0) return e1 / e2;
                    else throw new CustomException("Division by zero!");
            }
            throw new CustomException("Invalid operation!");
        }

        public String Tostring() { return expression1.Tostring() + op + expression2.Tostring(); }
    }
}
